---
title: "20240310 mouri et al. replication"
output: html_document
date: "2024-03-10"
---

Load the packages
```{r}
library(readxl)
library(data.table)
library(stringr)
library(IRanges)
library(SummarizedExperiment)
library(ggplot2)
library(pheatmap)
library(ChIPseeker)
# library(clusterProfiler)
library(reshape2)
library(TxDb.Hsapiens.UCSC.hg19.knownGene)
library(RColorBrewer)
library(ggrepel)
library(R.utils)
library(pegas)
library(gt)
```


```{r}
mpra1<-read.delim(paste0("/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/20240310_mpra_analysis/tcell/glm/data/20240310_tcell_glm_mpra_merge_hg38.txt"), header=T, stringsAsFactors = F, sep="\t")
mpra2<-read.delim(paste0("/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/20240310_mpra_analysis/unstim_jurkat/glm/data/20240310_unstim_jurkat_glm_mpra_merge_hg38.txt"), header=T, stringsAsFactors = F, sep="\t")
mpra3<-read.delim(paste0("/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/20240310_mpra_analysis/stim_jurkat/glm/data/20240310_stim_jurkat_glm_mpra_merge_hg38.txt"), header=T, stringsAsFactors = F, sep="\t")

# 4-6 Give a name to each MPRA merge file 
mpra1_name <- "20240310 Stimulated Tcell glm"
mpra2_name<- "20240310 Unstimulated Jurkat glm" 
mpra3_name<- "20240310 Stimulated Jurkat glm" 

# 7-9 Colors of venn diagram (This is optional)
color1 <- "turquoise2"
color2 <- "springgreen2"
color3 <- "forestgreen"

# 10 Title for the venn diagram
title <- "Variants with high activity and differential expression between alleles (emVars)"

# 11. plots.dir is the place where your plots are stored 
plots.dir <- "/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/20240310_mpra_analysis/plots"
# 12. plots_id is the prefix to the plot. It should end with an underscore to look nice
plots_id <- "20240310_mpra_compare_" 

mpra_venn_diagram(mpra1=mpra1, mpra2=mpra2, mpra3=mpra3, mpra1_name=mpra1_name, mpra2_name=mpra2_name, mpra3_name=mpra3_name, color1=color1, color2=color2,color3=color3,title=title, plots.dir=plots.dir, plots_id=plots_id)
```



The first input will be the MPRA merge file
```{r}
mpra.dir <- "/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/20240310_mpra_analysis/tcell/glm/data/20240310_tcell_glm_mpra_merge_hg38.txt"
mpra<-read.delim(paste0(mpra.dir), header=T, stringsAsFactors = F, sep="\t")

# An input in every function is going to be a subtitle which appears on all of the graphs to indicate which analysis you are preforming. This subtitle is a description in quotes and here we make it an object so we can put it in all of the plots. There will also be inputs for the plots directory and the plots identifier. 
subtitle <- "Tcell glm 03/10/24 Primary Tcell filter"
# plots.dir is the place where your plots are stored 
plots.dir <- "/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/20240310_mpra_analysis/tcell/glm/plots"
# plots_id is the prefix to the plot. It should end with an underscore to look nice
plots_id <- "20240310_tcell_glm" 
```


```{r}
mpra<-read.delim(paste0(mpra.dir), header=T, stringsAsFactors = F, sep="\t")
df <- data.frame(cbind(
  c(nrow(subset(mpra, mpra_sig == "Enhancer_Skew")),
    nrow(subset(mpra, mpra_sig == "Enhancer_nSkew")),
    nrow(subset(mpra, mpra_sig == "nEnhancer_nSkew")))),
  row.names = c("emVars","pCREs only","No Activity"))
# Give the data frame the proper headings
  colnames(df) <- c("Tcell MPRA")
  gtable_compare <- gt(df, rownames_to_stub = TRUE)
  gtable_compare
```







## Volcano plots
```{r}
# Error in if (abs(mpra[i, ]$A.log2FC) > abs(mpra[i, ]$B.log2FC)) { : missing value where TRUE/FALSE needed

# Fixed by this:
mpra2 <- mpra
mpra2$A.log2FC[is.na(mpra2$A.log2FC)]<- (0)
mpra2$B.log2FC[is.na(mpra2$B.log2FC)]<- (0)
mpra2$A.logPadj_BH[is.na(mpra2$A.logPadj_BH)]<- (0)
mpra2$B.logPadj_BH[is.na(mpra2$B.logPadj_BH)]<- (0)
mpra2$A.logPadj_BF[is.na(mpra2$A.logPadj_BF)]<- (0)
mpra2$B.logPadj_BF[is.na(mpra2$B.logPadj_BF)]<- (0)      

mpra_volcano_plots(mpra2, subtitle, plots.dir, plots_id)
```


```{r}
mpra_pics(mpra,emvars_or_not="F" ,subtitle, plots.dir, plots_id,cell_type="Tcell")
```


## dhs overlap
Instructions from the R file:
Download DHS data from https://zenodo.org/record/3838751#.X69tfEJKg6U
DHS_Index_and_Vocabulary_hg19_WM20190703.txt.gz
DHS_Index_and_Vocabulary_metadata.tsv
dat_bin_FDR01_hg19.txt.gz
```{r}
# 1. merge mpra data (must have chr, snp_start, snp_end, mpra_sig columns)

# 2. dhs.pos

# HG38
dhs.pos<- read.delim(gzfile("/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/20240310_mpra_analysis/data/hg38_dhs/DHS_Index_and_Vocabulary_hg38_WM20190703.txt.gz"), header=T, stringsAsFactors = F)

# 3. sample.dat
sample.dat<-read.delim("/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/20240310_mpra_analysis/data/hg38_dhs/DHS_Index_and_Vocabulary_metadata.tsv", sep="\t", header=T, stringsAsFactors = F)

# bring in dhs data
dhs.dat<-fread("/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/20240310_mpra_analysis/data/hg38_dhs/dat_bin_FDR01_hg38.txt.gz")

mpra_dhs_overlap(mpra, dhs.pos, sample.dat, dhs.dat, subtitle, plots.dir, plots_id)
```















#######################

hg19

########################


```{r}
mpra.dir <- "/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/20240310_mpra_analysis/tcell/glm/data/20240310_tcell_glm_mpra_merge_hg19.txt"
mpra<-read.delim(paste0(mpra.dir), header=T, stringsAsFactors = F, sep="\t")

subtitle <- "Tcell glm 03/10/24 Primary Tcell filter hg19"
# plots.dir is the place where your plots are stored 
plots.dir <- "/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/20240310_mpra_analysis/tcell/glm/plots"
# plots_id is the prefix to the plot. It should end with an underscore to look nice
plots_id <- "20240310_tcell_glm_hg19" 
```


```{r}
# 1. merge mpra data (must have chr, snp_start, snp_end, mpra_sig columns)

# 2. dhs.pos
# HG19
dhs.pos<- read.delim(gzfile("/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/2020_mouri_public_data_replication/data/original_hg19_dhs_data/DHS_Index_and_Vocabulary_hg19_WM20190703.txt.gz"), header=T, stringsAsFactors = F)

# 3. sample.dat
sample.dat<-read.delim("/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/2020_mouri_public_data_replication/data/original_hg19_dhs_data/DHS_Index_and_Vocabulary_metadata.tsv", sep="\t", header=T, stringsAsFactors = F)

# HG19
# bring in dhs data
dhs.dat<-fread("/nfs/jray/screens/ALL_MPRAs/Ho_et_al_analysis/Downstream_Analysis/2020_mouri_public_data_replication/data/original_hg19_dhs_data/dat_bin_FDR01_hg19.txt.gz")
#Subset for only DHS peaks that overlap with MPRA in the dhs data

# 5. yaxislimit (I have often changed the yaxis limit on this plot)
# yaxislimit_test <- 5.0

mpra_dhs_overlap(mpra, dhs.pos, sample.dat, dhs.dat, subtitle, plots.dir, plots_id)
```
```



## Allelic Skew
```{r}
dat.skew_file<-data.frame(read_excel("/nfs/jray/screens/Mouri_et_al_MPRA/asc/41588_2019_505_MOESM3_ESM.xlsx", sheet=10), stringsAsFactors = F)

mpra_allelic_skew(mpra=mpra, dat.skew=dat.skew_file, stimulation="S", subtitle=subtitle, plots.dir=plots.dir, plots_id=plots_id)
```

## TSS
```{r}
txdb <- TxDb.Hsapiens.UCSC.hg19.knownGene

mpra_tss(mpra=mpra,txdb=txdb, subtitle=subtitle, plots.dir=plots.dir, plots_id=plots_id)
```

## caQTL
```{r}
# 1. merge mpra data (must have chr, snp_start, snp_end, LogSkew, mpra_sig columns)

# 2. dat.qtl (From Gate et al ATAC-QTL data (PMID: 29988122), Supplemental Table 6)
dat.qtl_test <-data.frame(read_excel("/nfs/jray/screens/Mouri_et_al_MPRA/qtl/41588_2018_156_MOESM8_ESM.xlsx", sheet=2), stringsAsFactors = F)

mpra_caqtl(mpra,dat.qtl_test, subtitle, plots.dir, plots_id)
```

## deltaSVM
```{r}
# 1. merge mpra data (must have chr, snp_start, snp_end, SNP, mpra_sig, LogSkew columns)

# 2. dat.deltasvm
# This is for primary t-cells
cd4 <- read.delim(paste0("/nfs/jray/screens/Mouri_et_al_MPRA/delta_svm/mpra_snps_E2_naive_CD4_deltaSVM.txt"), header=F, stringsAsFactors = F, sep="\t")
# This is for jurkat cells
jurkat <-read.delim(paste0("/nfs/jray/screens/Mouri_et_al_MPRA/delta_svm/mpra_snps_E2_Jurkat_deltaSVM.txt"), header=F, stringsAsFactors = F, sep="\t")

# 3. subtitle 
# 4. plots.dir
# 5. plots_id

mpra_deltasvm(mpra, cd4, subtitle, plots.dir, plots_id)
```


## gwas loci counts
```{r}
# 1. merge mpra data (must have chr, snp_end,ld_snp, mpra_sig columns)

# 2. mpra.chr.dir (the directory of the mpra.chr files which are created through plink the end must be "mpra.chr")
mpra.chr.dir_test <- "/nfs/jray/screens/Mouri_et_al_MPRA/mhguo1-T_cell_MPRA-5c36361/annotate_mpra/ld/ld/mpra.chr"

mpra_gwas_loci_counts(mpra,mpra.chr.dir_test, subtitle, plots.dir, plots_id)
```

## gwas loci dhs
```{r}
# 1. merge mpra data
 
# 2. dat.alleles
#This file contains one row for each emVar
#"risk_allele" column shows whether the ref (A) or alt (B) allele is the risk allele
#Remaining columns show for each disease, whether the alt (B) allele increases or decreases risk for that disease. If not associated, then it's NA
dat.alleles_test<-data.frame(read_excel(paste0( "/nfs/jray/screens/Mouri_et_al_MPRA/gwas_loci/mpra_allele_annotation.xlsx"), sheet=1), stringsAsFactors = F)

# 3. a y axis limit for labeling the variants with numbers
textylimit_test <- 5

# 4. subtitle
# 5. plots.dir
# 6. plots_id
mpra_gwas_loci_dhs(mpra, dat.alleles_test, textylimit_test , subtitle, plots.dir, plots_id)
```
## Histone chromhmm
```{r}
# 1. merge mpra data (must have chr, snp_start, snp_end, ID, snp_sig and dhs_Tcell_merged columns)

# 2. manifest.histone (manifest file containing file paths to ENCODE histone ChIP bed files for T cells)
manifest.histone<-data.frame(read_excel("/nfs/jray/screens/Mouri_et_al_MPRA/mhguo1-T_cell_MPRA-5c36361/histone/histone_manifest.xlsx", sheet=1), stringsAsFactors = F)

# 3. cage
# I cannot download into R directly from the internet so I am going to download from the browser and upload to the server
# cage<-data.frame(fread("https://fantom.gsc.riken.jp/5/datafiles/latest/extra/Enhancers/human_permissive_enhancers_phase_1_and_2.bed.gz")) #read in CAGE data from FANTOM consortium
# Importing the data into R from the file which is now on the computer
cage<-data.frame(fread("/nfs/jray/screens/Mouri_et_al_MPRA/mhguo1-T_cell_MPRA-5c36361/histone/human_permissive_enhancers_phase_1_and_2.bed.gz"))

# 4. chromHMM file from ENCODE.
chromhmm<-fread("https://egg2.wustl.edu/roadmap/data/byFileType/chromhmmSegmentations/ChmmModels/core_K27ac/jointModel/final/E034_18_core_K27ac_dense.bed.gz", data.table = F, stringsAsFactors = F, header=F)

mpra_histone_chromhmm(mpra, manifest.histone, cage, chromhmm,  subtitle, plots.dir, plots_id)
```









